Wrong template parameter in destructor

*Originally posted by @jasonivey in https://github.com/ricab/scope_guard/pull/17#issue-1193795050*

There are two errors/warnings which seem to conflict:
* GCC 11 with C++20 enabled gives the error:
  ```
  scope_guard/scope_guard.hpp:152:36: error: template-id not allowed for destructor
    152 | sg::detail::scope_guard<Callback>::~scope_guard<Callback>() noexcept
          |                                    ^
  ```
* Clang 12/13/14 triggers the warning `-Wdtor-name`:
  ```
  warning: ISO C++ requires the name after ‘::~’ to be found in the same scope as the name before ‘::~’
  ```
  Note: this was (incorrectly) fixed with pull request #10 right before the last release

--- a/scope_guard.hpp
+++ b/scope_guard.hpp
@@ -149,7 +149,8 @@ noexcept(std::is_nothrow_constructible<Callback, Callback&&>::value)
 
 ////////////////////////////////////////////////////////////////////////////////
 template<typename Callback>
-sg::detail::scope_guard<Callback>::~scope_guard() noexcept
+sg::detail::scope_guard<Callback>::scope_guard::~scope_guard() noexcept  /*
+need the extra injected-class-name here to make different compilers happy */
 {
   if(m_active)
     m_callback();

--- a/scope_guard.hpp
+++ b/scope_guard.hpp
@@ -149,7 +149,7 @@ noexcept(std::is_nothrow_constructible<Callback, Callback&&>::value)
 
 ////////////////////////////////////////////////////////////////////////////////
 template<typename Callback>
-sg::detail::scope_guard<Callback>::~scope_guard<Callback>() noexcept
+sg::detail::scope_guard<Callback>::~scope_guard() noexcept
 {
   if(m_active)
     m_callback();

